common/lib/capa/capa/capa_problem.py

```diff
@@ -492,6 +495,121 @@ def get_answer_ids(self):
             answer_ids.append(results.keys())
         return answer_ids
 
+    def find_question_label(self, answer_id):
+        """
+        Obtain the most relevant question text for a particular answer.
+
+        E.g. in a problem like "How much is 2+2?" "Two"/"Three"/"More than three",
+        this function returns the "How much is 2+2?" text.
+
+        It uses, in order:
+        - the question prompt, if the question has one
+        - the <p> or <label> element which precedes the choices (skipping descriptive elements)
+        - a text like "Question 5" if no other name could be found
+
+        Arguments::
+            answer_id: a string like "98e6a8e915904d5389821a94e48babcf_13_1"
+
+        Returns:
+            a string with the question text
+        """
+        # Some questions define a prompt with this format:   >>This is a prompt<<
+        assert answer_id in self.problem_data
+        prompt = self.problem_data[answer_id].get('label')
+
+        if prompt:
+            question_text = prompt.striptags()
+        else:
+            # If no prompt, then we must look for something resembling a question ourselves
+            #
+            # We have a structure like:
+            #
+            # <p />
+            # <optionresponse id="a0effb954cca4759994f1ac9e9434bf4_2">
+            #   <optioninput id="a0effb954cca4759994f1ac9e9434bf4_3_1" />
+            # <optionresponse>
+            #
+            # Starting from  answer (the optioninput in this example) we go up and backwards
+            xml_elems = self.tree.xpath('//*[@id="' + answer_id + '"]')
```

Comment:
It makes much sense but it's not strictly „accessibility“ and I don't understand yet all cases that this function handles (multiple inputfields, deleting inputs, …). I fear there won't be time to do this. I added a TODO and I'll still try to do it because it makes sense. `response_a11y_data` some related (though different) cases to the ones I did.

Reply:
I say that since anything we find that we consider a question label, will also be useful as an ARIA description for a screen reader. If it isn't useful in that sense, chances are it's also not very useful as a question label.

Reply:
I'm okay with keeping this separate from `response_a11y_data` for now; I think in the future it'd make sense to have that function call into this one.